pageserver/src/tenant.rs

```diff
@@ -1214,12 +1340,49 @@ impl Tenant {
                     )
                 }
                 Err(DownloadError::NotFound) => {
-                    info!("no index file was found on the remote");
+                    info!("no index file was found on the remote, found_delete_mark: {found_delete_mark}");
+
+                    if found_delete_mark {
+                        // We could've resumed at a point where remote index was deleted, but metadata file wasnt.
+                        // Cleanup:
+                        self.cleanup_remaining_fs_traces_after_timeline_deletion(timeline_id)
+                            .await
+                            .context("cleanup_remaining_fs_traces_after_timeline_deletion")
+                            .map_err(LoadLocalTimelineError::resume_delete)?;
+                    }
+
+                    // We're loading fresh timeline that didnt yet make it into remote.
+                    // FIXME: now we block on uploading index during both branch creation and initial creation, is is considered normal?
```

Comment:
If this is related to me fixing #4204 then I'd say it's considered normal. But maybe I misunderstand, because I don't see how this is related.

Reply:
Removed in [930ea1c](https://github.com/neondatabase/neon/pull/4552/commits/930ea1c8dff763cfbfd25c5842337aa1b2b47898) but would be interested to hear your inkling of a thought regardless or was it just a misunderstanding?

Reply:
Well, maybe it was just the blocking or waiting for upload consideration once more. Makes me think if there should be more comments around the work, but the comments are in the caller if I understand this site correctly.

Reply:
Yep, I was thinking how to treat those events, should there be a warning log or something. But you're right, we can still get there without anything strange going on, so this is part of the normal flow so no warning is needed.

Was also thinking about retries of `create_timeline` calls. But they're also ok because of this wait_completion call: https://github.com/zenithdb/zenith/blob/930ea1c8dff763cfbfd25c5842337aa1b2b47898/pageserver/src/tenant.rs#L1539